libass/ass_bitmap_engine.h

```diff
@@ -87,6 +87,7 @@ enum {
     ASS_CPU_FLAG_X86_AVX512ICL = 0x04,  // placeholder for checkasm
     ASS_CPU_FLAG_MAX,
     ASS_CPU_FLAG_ALL = ((ASS_CPU_FLAG_MAX & ~1ull) << 1) - 1,
+    ASS_FLAG_LARGE_TILES       = 0x08,
```

Comment:
That reminds me: why is it `ASS_CPU_FLAG_MAX & ~1ull` rather than the simpler `ASS_CPU_FLAG_MAX - 1ull`?

> Do we actually need a CPU-specfic `ASS_CPU_FLAG_ALL`, or could we use a general `ASS_FLAG_ALL`-mask instead?

Well, the way this is used in this PR is that `ASS_CPU_FLAG_ALL | optionally, depending on configure flags, ASS_FLAG_LARGE_TILES` are enabled. But I’m not sure what the goal is.

Reply:
> each addition of a new CPU-related flag will also need to modify the values of non-CPU flags

Good catch. I'll add some room for them, i.&nbsp;e. `ASS_FLAG_LARGE_TILES = 0x10000` or something.

> Do we actually need a CPU-specfic `ASS_CPU_FLAG_ALL`, or could we use a general `ASS_FLAG_ALL`-mask instead?

No, it's exactly what is needed. It's intention is "use all CPU features available". All other flags are configuration, we want to enable them individually.

> For my own understanding, can you elaborate on why this is needed?

Look into my last [checkasm](https://github.com/MrSmile/libass/commits/checkasm) branch, specifically [here](https://github.com/MrSmile/libass/commit/83a8f169481e3a5c2a73de93ee51db62565f3471#diff-b878f925440d48c88a81a7564472974abb7170db8259751a8d5de111c3833accR285).

> That reminds me: why is it `ASS_CPU_FLAG_MAX & ~1ull` rather than the simpler `ASS_CPU_FLAG_MAX - 1ull`?

I think I'll remove `ASS_CPU_FLAG_X86_AVX512ICL`, `ASS_CPU_FLAG_MAX` and redefine `ASS_CPU_FLAG_ALL` to a fixed number like `0xFFFF`.

Reply:
> I'll add some room for them, i. e. `ASS_FLAG_LARGE_TILES = 0x10000` or something. 

ISO-C99 only guarantees enums to hold the same values as `int` and `int`'s maximum value must only be at least `2^15 - 1`, so `0x10000` is technically UB, depending on the target platform. *(POSIX raises the minimal acceptable value to `2^31 - 1`)*
Iinm currently all of libass' enum values are below `2^15 - 1`.

>  Look into my last checkasm branch, specifically here.

Thanks.

Reply:
Standard reminder that the rest of libass assumes `int` holds 32 bits anyway, as much as I don’t like it. But this has been improving with MrSmile’s PRs that adopt fixed-width types, and it shouldn’t hurt to make this enum more compliant ahead of time—but only if we think it leaves enough bits for future expansion, of course.